src/test/scala/chisel3/tests/BundleLiteralsSpec.scala

```diff
@@ -36,35 +25,44 @@ class BundleLiteralSpec extends FlatSpec with ChiselScalatestTester {
       io.aOut := io.in.a
       io.bOut := io.in.b
     }) { c =>
-      c.io.in.poke(c.io.in.Lit(0.U, 1.U))
+      c.io.in.poke(c.io.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
       c.io.aOut.expect(0.U)
       c.io.bOut.expect(1.U)
 
-      c.io.in.poke(c.io.in.Lit(2.U, 5.U))
+      c.io.in.poke(c.io.in.cloneType.Lit(_.a -> 2.U, _.b -> 5.U))
       c.io.aOut.expect(2.U)
       c.io.bOut.expect(5.U)
     }
   }
 
   it should "expect Bundle literals" in {
     test(new PassthroughModule(new DoubleElements)) { c =>
-      c.in.poke(c.in.Lit(0.U, 1.U))
-      c.out.expect(c.in.Lit(0.U, 1.U))
-      c.in.poke(c.in.Lit(2.U, 5.U))
-      c.out.expect(c.in.Lit(2.U, 5.U))
+      c.in.poke(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
+      c.out.expect(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
+      c.in.poke(c.in.cloneType.Lit(_.a -> 2.U, _.b -> 5.U))
+      c.out.expect(c.in.cloneType.Lit(_.a -> 2.U, _.b -> 5.U))
     }
   }
 
   it should "fail on expect mismatch" in {
     assertThrows[exceptions.TestFailedException] {
       test(new PassthroughModule(new DoubleElements)) { c =>
-        c.in.poke(c.in.Lit(0.U, 1.U))
-        c.out.expect(c.in.Lit(0.U, 2.U))
+        c.in.poke(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
+        c.out.expect(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 2.U))
       }
     }
   }
 
-  ignore should "peek Bundle literals" in {
-    // TODO to be written
+  // peek on bundle not supported yet, this test will fail and should
```

Comment:
Why'd you get rid of the ignore? I think the ignore is cleaner and expresses intent better than an exception intercept. The test name is also very confusing.

Reply:
I think getting rid of the ignore is a better way of making sure we update the test when peeking is of Bundled is enabled. I agree the test name is bad and is [changed.](url)

Reply:
I'd prefer expressing intent over some confusing system that results in automated reminders. Because when you're building a new feature, you should already be thinking of writing tests for it, and the ignored should act as a flag that gets cleaned up (though perhaps eventually). And the tests are intended to essentially codify behavior, and a LiteralTypeException is not the end goal here.

I guess ideally there would be an ignore flag that checks for a test failure and suppresses it, but errors on a non-failure. Don't know if that exists in ScalaTest.

Reply:
Test is now an ignore